torch/nn/modules/container.py

```diff
@@ -50,14 +50,21 @@ def __init__(self, *args):
                 self.add_module(str(idx), module)
 
     def __getitem__(self, idx):
-        if not (-len(self) <= idx < len(self)):
-            raise IndexError('index {} is out of range'.format(idx))
-        if idx < 0:
-            idx += len(self)
-        it = iter(self._modules.values())
-        for i in range(idx):
-            next(it)
-        return next(it)
+        if isinstance(idx, slice):
+            ret_seq = Sequential()
```

Comment:
Why not just `return Sequential(*list(self._modules)[idx])`? To preserve named modules?

Reply:
Yes. I just assumed that a slice of `Sequential` would require properties of the named modules to be retained.